Packages/HBPizzaHelperAPI/Sources/HBPizzaHelperAPI/HTTPRequest.swift

```diff
@@ -0,0 +1,486 @@
+                                error.localizedDescription + "\n"
+                        )
+                    } else {
+                        guard let data = data else {
+                            completion(.failure(.noResponseData))
+                            print("found response data nil")
+                            return
+                        }
+                        guard response is HTTPURLResponse else {
+                            completion(.failure(.responseError))
+                            print("response error")
+                            return
+                        }
+                        DispatchQueue.main.async {
+                            guard let stringData = String(
+                                data: data,
+                                encoding: .utf8
+                            ) else {
+                                completion(
+                                    .failure(
+                                        .decodeError(
+                                            "fail convert data to .utf8 string"
+                                        )
+                                    )
+                                ); return
+                            }
+                            print(stringData)
+                            let data = stringData.replacingOccurrences(
+                                of: "\"NaN\"",
+                                with: "0"
+                            ).data(using: .utf8)!
+                            let decoder = JSONDecoder()
+                            if baseHost != "http://81.70.76.222" {
+                                decoder
+                                    .keyDecodingStrategy = .convertFromSnakeCase
+                            }
+                            guard let response = response as? HTTPURLResponse
+                            else {
+                                completion(.failure(.responseError))
+                                return
+                            }
+                            print(response.statusCode)
+
+                            do {
+                                let requestResult = try decoder.decode(
+                                    T.self,
+                                    from: data
+                                )
+                                completion(.success(requestResult))
+                            } catch {
+                                print(error)
+                                completion(.failure(.decodeError(
+                                    error
+                                        .localizedDescription
+                                )))
+                            }
+                        }
+                    }
+                }.resume()
+            }
+        } else {
+            completion(.failure(.responseError))
+        }
+    }
+}
+
+// String的扩展，让其具有直接加键值对的功能
+extension String {
```

Comment:
Do not use this extension. Use `URLQueryItem` to build `URL` then send request instead.